### PR TITLE
Apply constraints from setUp if any

### PR DESCRIPTION
**Pull request by iFrostizz**
Closes: https://github.com/runtimeverification/evm-semantics/issues/1878

Minimal reproduction: https://github.com/iFrostizz/setup-constraints-repro

### CODE REVIEW

**Comment by ehildenb** [comment]
@iFrostizz please back out the changes to the K submodule from your PR. In general, _always_ let our update automation handle updating dependencies and submodules. Otherwise, the next time the automation tries to make an update, it will be confused.

File: kevm-pyk/src/kevm_pyk/foundry.py [hunk 1]
```diff
@@ -886,28 +885,22 @@ def _method_to_cfg(
...
-    return init_cterm
-
-
-def get_final_accounts_cell(proof_digest: str, proof_dir: Path) -> tuple[KInner, KInner]:
+def get_final_accounts_cell(proof_digest: str, proof_dir: Path) -> tuple[KInner, KInner, tuple[KInner, ...]]:
     apr_proof = APRProof.read_proof(proof_digest, proof_dir)
     target = apr_proof.kcfg.get_unique_target()
     cterm = single(apr_proof.kcfg.covers(target_id=target.id)).source.cterm
-    return (cterm.cell('ACCOUNTS_CELL'), cterm.cell('ACTIVEACCOUNTS_CELL'))
+    return (cterm.cell('ACCOUNTS_CELL'), cterm.cell('ACTIVEACCOUNTS_CELL'), cterm.constraints)
```
[the remark this answers is not in the document]
Comment (ehildenb) [reply kevm-pyk/src/kevm_pyk/foundry.py:892]: This functionality can probably be factored out into a generic method in `pyk.kast.manip` which looks like:

```
def constraints_for(vars: Iterable[str], constraints: Iterable[KInner]) -> Iterable[KInner]: ...
```

**Comment by ehildenb** [comment]
@iFrostizz make sure to add the minimal test to the test-suite in `tests/foundry/...`, so that we know it's working on CI.

**Comment by iFrostizz** [comment]
Refactor done and test added.

**Review by ehildenb — APPROVED** [review]

File: tests/foundry/test/Preconditions.t.sol
[the remark this answers is not in the document]
Comment (iFrostizz) [reply tests/foundry/test/Preconditions.t.sol:1]: Thank you for noticing, added it in https://github.com/runtimeverification/evm-semantics/pull/1924

**Pull request merged**